OpenSim/Common/TableSource.h

```diff
@@ -95,21 +121,52 @@ class TableSource_ : public Component {
         return _table;
     }
 
-    /** Replace the existing TimeSeriesTable_ this TableSource_ currently 
-    holds.                                                                    
+    /** Replace the existing TimeSeriesTable_ that this TableSource_ currently 
+    holds. This operation is not allowed if TableSource_ already contains a
+    value for the property `filename`. In such cases, update the property
+    'filename' using `upd_filename()` to empty string and call this function.
 
-    \throws KeyNotFound If table provided does not have column-labels.        */
+    \throws InvalidCall If property `filename` is set. This call is not allowed
+                        if `filename` property is set.                        
+    \throws KeyNotFound If table provided does not have column labels.        */
     void setTable(const Table& table) {
+        OPENSIM_THROW_IF(!get_filename().empty(),
+                         InvalidCall,
+                         "Property 'filename' is set. Cannot set Table now.");
```

Comment:
I have changed the functions to not throw. I think the only argument for it to be consistent with existing operations.

I am in favor of throwing an exception during a call to read from a filename when a value for filename already exists, instead of resetting the filename during the operation. 
* It can lead to mistakes (as Chris points out). 
* It also makes the code unreadable because a reader has to go through the documentation of every operation to know if properties are being changed. If code includes an explicit reset of the filename, then reader will can see clearly.
* As a result of above, debugging becomes harder if one desires to track a property.